gcp_variant_transforms/pipeline_common_test.py

```diff
@@ -41,24 +41,46 @@ def _create_mock_args(self, **args):
   def _get_pipeline_mode(self, args):
     all_patterns = pipeline_common._get_all_patterns(args.input_pattern,
                                                      args.input_file)
-    return pipeline_common.get_pipeline_mode(all_patterns)
+    return pipeline_common.get_pipeline_mode(all_patterns,
+                                             args.pipeline_mode)
 
   def test_validation_failure_for_invalid_input_pattern(self):
     with self.assertRaisesRegex(
         ValueError, 'Input pattern .* did not match any files.'):
       pipeline_common._get_all_patterns(
           input_pattern='nonexistent_file.vcf', input_file=None)
 
+  def test_get_mode_small_pipeline_mode(self):
+    args = self._create_mock_args(input_pattern='*', input_file=None, pipeline_mode='small')
+    match_result = collections.namedtuple('MatchResult', ['metadata_list'])
+    match = match_result([None for _ in range(100)])
+    with mock.patch.object(FileSystems, 'match', return_value=[match]):
+      self.assertEqual(self._get_pipeline_mode(args), PipelineModes.SMALL)
+
+  def test_get_mode_medium_pipeline_mode(self):
+    args = self._create_mock_args(input_pattern='*', input_file=None, pipeline_mode='medium')
+    match_result = collections.namedtuple('MatchResult', ['metadata_list'])
+    match = match_result([None for _ in range(100)])
+    with mock.patch.object(FileSystems, 'match', return_value=[match]):
+      self.assertEqual(self._get_pipeline_mode(args), PipelineModes.MEDIUM)
+
+  def test_get_mode_large_pipeline_mode(self):
+    args = self._create_mock_args(input_pattern='*', input_file=None, pipeline_mode='large')
+    match_result = collections.namedtuple('MatchResult', ['metadata_list'])
+    match = match_result([None for _ in range(100)])
+    with mock.patch.object(FileSystems, 'match', return_value=[match]):
+      self.assertEqual(self._get_pipeline_mode(args), PipelineModes.LARGE)
+
   def test_get_mode_small(self):
-    args = self._create_mock_args(input_pattern='*', input_file=None)
+    args = self._create_mock_args(input_pattern='*', input_file=None, pipeline_mode='')
```

Comment:
totally a nit: you don't need to pass `pipeline_mode` at all here as the argument is defaulted to the same, and this test (and the next few) ensure existing behavior continues, right?

Reply:
Yes, the tests are to ensure both existing and new behavior. The `pipeline_mode` needs to be passed in for test setup.